libraries/appbase/include/appbase/application.hpp

```diff
@@ -65,14 +65,21 @@ class application_factory
       }
    }
 
-   application& new_application( const string& id);
+   application& new_application( const string& id );
+
+   application& new_application( const string& id, application app );
+
+   application& get_application( const string& id ){
+      return apps.at(id);
+   }
 
    template<typename Plugin>
    std::shared_ptr< abstract_plugin > new_plugin(){
       return new_plugin(Plugin::name());
    }
 
    std::shared_ptr< abstract_plugin > new_plugin(string name){
+      assert(plugin_factories.count(name));
```

Comment:
Why is id public member and not private ? 
https://github.com/SophiaTX/SophiaTX/pull/223/files#diff-7970c9966f3dba8380b0be41a9e433faR215

Reply:
It is used in some of the macros during API registration

Reply:
But I made it const, to make it clear it is set only in constructor